frontend/src/hooks/useHashURL.js

```diff
@@ -17,17 +17,12 @@ import { useCallback } from "react"
  * @param {*} defaultKey 
  * @returns 
  */
+
 export const useHashURL = (defaultKey) => {
     const history = useNavigate()
     const location = useLocation()
     const currentKey = location.hash.slice(1)
-    const setCurrentKey = useCallback((key) => { history(`#${key}`) }, [])
```

Comment:
useCallback was causing a routing issue where clicking on a node would properly update the current url with the new sample id, but clicking on a tab in sample details would "reset" the sample id to a previous sample, and this would occasionally cause the lineage graph to crash if it couldn't find the sample in samplesById.
The problem is that I used useCallback without specifying any dependencies. I thought that `history` was just a function, but in fact it is a closure and a location is stored in the closure scope. My callback had a closure on a stale version of history that contained a stale url, and calling `history` would reset the url to a previously visited sample.
I don't think useCallback adds any real value here, so I just got rid of it.